src/resources/scripts/countRequests.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+nbrVisits=0
+nbrActions=0
+
+while IFS= read -r line; do
+    # ignore bot and security check requests
+    bot="[bB]ot|OpenVAS-VT";
+    if [[ "$line" =~ $bot ]]; then
+        continue;
+    fi
+    
+    browser="Chrome|Edg|Safari|OPR|Firefox";
+    if [[ "$line" =~ $browser ]]; then
+    # filter http code (2xx) by using regex of ' <http code> XY ' 
+      code="[[:space:]]2[[:digit:]]{2}[[:space:]][[:digit:]]+[[:space:]]"
```

Comment:
It may be safer to match the code and HTTP verb by their position as determined by the space separators. Example:
```suggestion
      code=$(echo "$line" | cut -d ' ' -f 9)
      verb=$(echo "$line" | cut -d ' ' -f 6 | cut -c 2-)
```

While the code regex looks reasonably specific, the verb could easily match an URL (e.g. `https://biigle.de/POSTER`).

Reply:
When I apply your changes and try to run the script, it takes 3-5 minutes to terminate. Is my machine just slow or does the time not matter in this case?

Reply:
If the script was faster before on your machine, it's slower now :smiley: Faster is better. I'll have a look.

Reply:
You are right, it's quite horrible with the change proposed above. This works quite fast:

```bash
#!/bin/bash
#
# Usage: ./countRequests.sh logfile.log.gz

TOKEN=""
BASE_URL="https://biigle.de"

compute_visits() {
    zgrep -F -e Chrome -e Firefox -e Safari -e OPR -e Edg $1 \
        | cut -d ' ' -f 6,9 \
        | cut -c 2-6 \
        | grep -F "GET 2" \
        | wc -l
}

compute_actions() {
    zgrep -F -e Chrome -e Firefox -e Safari -e OPR -e Edg $1 \
        | grep -F -v heartbeat \
        | cut -d ' ' -f 6 \
        | cut -c 2- \
        | grep -F -e PUT -e POST -e DELETE \
        | wc -l
}

nbrVisits=$(compute_visits $1)
nbrActions=$(compute_actions $1)

echo curl \
    -X POST \
    -H "Authorization: Bearer $TOKEN" \
    -H "Content-Type: application/json" \
    -d "{\"visits\": $nbrVisits, \"actions\": $nbrActions}" \
    ${BASE_URL}/api/v1/kpis

```

It also works with the gzipped logfiles and ignores the automatic "heartbeat" action.

Reply:
Thinking about it, the script above does not report the numbers described in #1. The visits should only count the dashboard/home page. The actions should also include automated API requests (via token authentication). Here is an updated script:

```bash
#!/bin/bash
#
# Usage: ./countRequests.sh logfile.log.gz

TOKEN=""
BASE_URL="https://biigle.de"

compute_visits() {
    zgrep -F -e Chrome -e Firefox -e Safari -e OPR -e Edg $1 \
        | cut -d ' ' -f 6,7,9 \
        | cut -c 2-8 \
        | grep -F "GET / 2" \
        | wc -l
}


compute_actions() {
    # The "@" will appear in API requests with token authentication.
    zgrep -F -e PUT -e POST -e DELETE $1 \
        | grep -F -e Chrome -e Firefox -e Safari -e OPR -e Edg -e "@" \
        | grep -F -v heartbeat \
        | wc -l
}

nbrVisits=$(compute_visits $1)
nbrActions=$(compute_actions $1)

curl \
    -X POST \
    -H "Authorization: Bearer $TOKEN" \
    -H "Content-Type: application/json" \
    -d "{\"visits\": $nbrVisits, \"actions\": $nbrActions}" \
    ${BASE_URL}/api/v1/kpis
```